pkg/steps/index_generator.go

```diff
@@ -0,0 +1,143 @@
+package steps
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	buildapi "github.com/openshift/api/build/v1"
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/results"
+	"github.com/openshift/ci-tools/pkg/util"
+	imageclientset "github.com/openshift/client-go/image/clientset/versioned/typed/image/v1"
+	coreapi "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type indexGeneratorStep struct {
+	config      api.IndexGeneratorStepConfiguration
+	resources   api.ResourceConfiguration
+	buildClient BuildClient
+	imageClient imageclientset.ImageStreamsGetter
+	istClient   imageclientset.ImageStreamTagsGetter
+	jobSpec     *api.JobSpec
+	artifactDir string
+	pullSecret  *coreapi.Secret
+}
+
+const IndexDataDirectory = "/index-data"
+const IndexDockerfileName = "index.Dockerfile"
+
+func (s *indexGeneratorStep) Inputs() (api.InputDefinition, error) {
+	return nil, nil
+}
+
+func (s *indexGeneratorStep) Run(ctx context.Context) error {
+	return results.ForReason("building_index_generator").ForError(s.run(ctx))
+}
+
+func (s *indexGeneratorStep) run(ctx context.Context) error {
```

Comment:
we might want to identify specific `results.ForReason()` with the errors coming out of this so it's not opaque -- which of these errors are infra-caused? All of them? Is it possible for any to be user-based?

Reply:
There is 1 situation where it could be user-based, which would be the bundle not containing certain info (like missing a CRD) or containing incorrect info. I'll add a regex to try and identify if that is the issue based on logs.

Reply:
Added check